### PR TITLE
Alerting: Start ticker only when scheduler starts

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What this PR does / why we need it**:
The current ticker implementation starts it when it is created and it immediately starts producing ticks. If ticks are not consumed from ticker it "piles up" ticks and then when ticks are consumed it tries to catch up.
The only consumer of the ticker is the scheduler (legacy or new one). The ngalert scheduler creates a ticker in the constructor 

https://github.com/grafana/grafana/blob/4163f31d7613df6b1a9e1e46d7ea57602d13c945/pkg/services/ngalert/schedule/schedule.go#L119-L120

However, there can be a very substantial gap between the scheduler is created and the scheduler is started (i.e. starts consuming ticks)
https://github.com/grafana/grafana/blob/ba97b268d0429b3f2adaf06ea04010ea590fba31/pkg/services/ngalert/schedule/schedule.go#L145

Therefore, if scheduler is started more than 10 seconds after it was created, it will start evaluating rules very fast until ticker catches up to the current time. 

One of those situations is the provisioning. Currently, it can take many minutes

This PR changes the time when ticker is initialized to when the scheduler starts. This will make the ticker to be consumed by the scheduler as fast as possible.

### CODE REVIEW

File: pkg/services/ngalert/schedule/schedule.go [hunk 1]
```diff
@@ -442,16 +438,6 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
...
 
-// overrideCfg is only used on tests.
-func (sch *schedule) overrideCfg(cfg SchedulerCfg) {
-	sch.clock = cfg.C
-	sch.baseInterval = cfg.Cfg.BaseInterval
-	sch.ticker.Stop()
-	sch.ticker = ticker.New(cfg.C, cfg.Cfg.BaseInterval, cfg.Metrics.Ticker)
-	sch.evalAppliedFunc = cfg.EvalAppliedFunc
-	sch.stopAppliedFunc = cfg.StopAppliedFunc
-}
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/schedule/schedule.go:453]: This is not used anymore.

**Review by JacobsonMT — APPROVED** [review]
LGTM :rocket:

**Comment by grobinson-grafana** [comment]
Should we have a `Start` method? Then we can still initialize the ticker in `NewScheduler`, but not start it until `Run`?

**Comment by yuri-tceretian** [comment]
> Should we have a Start method? Then we can still initialize the ticker in NewScheduler, but not start it until Run?

This solution works well. I do not think we need that.

**Pull request merged**